app/assets/javascripts/proxy_status.js

```diff
@@ -14,7 +14,11 @@ $(document).on('ContentLoad', function() {
 $(window).on('hashchange', setTab); //so buttons that link to an anchor can open that tab
 
 function setItemStatus(item, response) {
-  if(response.success) {
+  if(response.success && response.message.warning) {
```

Comment:
if `response.message` could be `undefined` then consider use lodash`._get` , otherwise it may cause errors.

Reply:
@amirfefer we are actually trying to reduce the usage of lodash, /cc @sharvit

Reply:
We do try to reduce the lodash usage, `_.get` can still be used in the assets but prefer not to whenever possible.
See: https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/bundle_lodash.js

Reply:
can do:
```js
if(response && response.success && response.message && response.message.warning) {}
```

Reply:
if `._get` exists in  global, why not using it in legacy js? even if it just for increasing readability a bit?

Reply:
I personally don't like the `_.get` and i think it takes away readability. (unless you are familiar with it)
But i am fine with that.

Reply:
If the response is successful, there always be a message node in the [response](https://github.com/theforeman/foreman/blob/develop/app/controllers/smart_proxies_controller.rb#L140).

Reply:
I don't believe the client should count on that.
I think it is still important to verify the object structure before you are using it.